projects/RabbitMQ.Client/client/impl/AsyncManualResetEvent.cs

```diff
@@ -117,10 +117,13 @@ public void Reset()
             _valueTaskSource.Reset();
         }
 
-        void IValueTaskSource.GetResult(short token) => _valueTaskSource.GetResult(token);
+        void IValueTaskSource.GetResult(short token) =>
```

Comment:
I think it would make sense to check the token against the version of the source and if it is not the same throw with a throw helper. But I can do this improvement also in another PR

Reply:
Might as well get it here, thanks.